core/build.gradle

```diff
@@ -49,66 +37,47 @@ dependencies {
     // Reactive Streams
     compile 'org.reactivestreams:reactive-streams'
 
-    // Internal dependencies which will be shaded by the 'internalDeps' task below.
-    internalDependency 'com.spotify:completable-futures'
-    internalDependency 'it.unimi.dsi:fastutil'
-    internalDependency 'org.reflections:reflections'
-
-    // Used by the 'relocateInternalDependencies', because ShadowTask doesn't let me specify a file.
-    trimmedInternalDependency files(internalDependenciesJarTrimmed)
+    // Reflections
+    compile 'org.reflections:reflections'
 }
 
-task trimInternalDependencies(type: ProGuardTask) {
-    injars files(configurations.internalDependency.collect()), filter: '!META-INF/**'
-    outjars internalDependenciesJarTrimmed
-    libraryjars files(configurations.runtime.collect() - configurations.internalDependency.collect())
+tasks.shadedJar.archiveName = "${tasks.jar.baseName}-untrimmed-${tasks.jar.version}.jar"
+
+task trimShadedJar(type: ProGuardTask) {
+    def shadedFile = tasks.shadedJar.archivePath
+    def shadedAndTrimmedFile = file(shadedFile.path.replaceFirst('-untrimmed-', '-shaded-'))
+    injars shadedFile
+    outjars shadedAndTrimmedFile
+    libraryjars files(configurations.runtime.collect())
     libraryjars file("${System.getProperty('java.home')}/lib/rt.jar")
 
     dontoptimize
     dontobfuscate
     dontwarn // Ignore the harmless 'missing classes' warnings related with the optional dependencies.
 
     keepattributes 'Signature, InnerClasses, Annotation'
+    keep 'class !com.linecorp.armeria.internal.**,com.linecorp.armeria.** { *; }'
```

Comment:
We have many source files in the internal package too and presumably want to always include, regardless of what proguard thinks.

I think we should put dependencies in a separate package - common names I've seen are relocated, repackaged, and shaded.